corehq/apps/locations/tests/test_location_sync.py

```diff
@@ -0,0 +1,76 @@
+from couchdbkit import ResourceNotFound
+from django.test import TestCase
+from corehq.apps.domain.shortcuts import create_domain
+from ..models import SQLLocation, Location
+from .util import setup_location_types, delete_all_locations
+
+DOMAIN = "location_sync_test"
+
+
+def couch_loc(name, location_type, parent=None):
+    loc = Location(site_code=name, name=name, domain=DOMAIN,
+                   location_type=location_type, parent=parent)
+    loc.save()
+    return loc
+
+
+def sql_loc(name, location_type, parent=None):
+    loc = SQLLocation(site_code=name, name=name, domain=DOMAIN,
+                      location_type=location_type, parent=parent)
+    loc.save()
+    return loc
+
+
+class TestLocationSync(TestCase):
```

Comment:
what i've typically tested for in addition to what's here is to:

- include an edit test case for each model (i.e., changing an already created object)
- compare counts during the tests after each sync to make sure that we're not creating too many records on create or edit, and that we're deleting all the records on delete
- set all values possible on the model so that the compare process compares everything gets migrated